urllib3/contrib/pyopenssl.py

```diff
@@ -167,13 +178,60 @@ def __init__(self, connection, socket, suppress_ragged_eofs=True):
         self.socket = socket
         self.suppress_ragged_eofs = suppress_ragged_eofs
         self._makefile_refs = 0
+        self._closed = False
 
     def fileno(self):
         return self.socket.fileno()
 
-    def makefile(self, mode, bufsize=-1):
-        self._makefile_refs += 1
-        return _fileobject(self, mode, bufsize, close=True)
+    if _fileobject:  # Platform-specific: Python 2
+        def makefile(self, mode, bufsize=-1):
+            self._makefile_refs += 1
+            return _fileobject(self, mode, bufsize, close=True)
+    else:  # Platform-specific: Python 3
```

Comment:
How do you envision this looking? `backports.makefile` exposing a pair of functions that I just assign, eg:

```python
try:  # Platform-specific: Python 2
    from socket import _fileobject
except ImportError:  # Platform-specific: Python 3
    _fileobject = None
    from .packages.backport_makefile import makefile, _decref_socketios as backport_makefile, backport_decref_socketios

class WrappedSocket(object):
    if _fileobject:
        def makefile(self, mode, bufsize=-1):
            self._makefile_refs += 1
            return _fileobject(self, mode, bufsize, close=True)
    else:
        makefile=backport_makefile
        _decref_socketios=backport_decref_socketios
```

That makes me a bit nervous because it requires that implementation details of the `WrappedSocket` class leak into the backport module, but it might be fine.